api/docs/bt.dox

```diff
@@ -1283,6 +1284,54 @@ load/store pair into a macro-instruction, which prevents any loads and
 stores from being inserted, but also prevents any of the instructions
 involved from being instrumented.
 
+***************************************************************************
+\htmlonly
+<table width=100% bgcolor="#000000" cellspacing=0 cellpadding=2 border=0>
+  <tr><td><table width=100% bgcolor="#0000C0" cellspacing=0 cellpadding=1 border=0>
+  <tr><td><table width=100% bgcolor="#0000C0" cellspacing=0 cellpadding=1 border=0>
+  <tr><td></td></tr></table></td></tr></table></td></tr></table>
+\endhtmlonly
+\section sec_rseq Restartable Sequence Instrumentation Constraints
+
+The Linux kernel supports special code regions called restartable
+sequences.  This "rseq" feature is challenging to support under
+instrumentation due to the tight restrictions on operations inside the
+sequence.  Instrumentation inserted in the sequence would need to be
+designed to be restartable as well, with a single commit point.  Meeting
+such requirements is unrealistic for most instrumentation.  Instead, DR
+provides a "run twice" solution where the sequence is first executed as
+regular code with regular instrumentation up to the commit point.  Then the
+sequence is restarted and executed without instrumentation to perform the
+commit.
+
+This run-twice approach is subject to the following limitations:
+
+- The application must store an rseq_cs struct for each rseq region in a
+  section of its binary named "__rseq_cs", optionally with an "__rseq_cs_ptr_array"
+  section of pointers into the __rseq_cs section, per established conventions.
+  These sections must be located in loaded segments.
+- Each rseq region's code must never be also executed as a non-restartable sequence.
+- Each rseq region must make forward progress if its abort handler is always
+  called the first time it is executed.
+- Each memory store instruction inside an rseq region must have no other side
+  effects: it must only write to memory and not to any registers.
```

Comment:
This has nothing to do with the abort handler.  This is a limitation in the initial implementation of turning a store into a nop: it simply removes the instruction.  Thus, it will not do the right thing if the store modifies registers.  E.g., it will fail to update the stack pointer for a push instruction.

Reply:
It seems quite clear to me.  I assume your misunderstanding was based on thinking it had to do with effects visible in the abort handler.

Reply:
@Carrotman42 did you have a further suggestion here?  Your discussion on side effects and state in the abort handler will become more relevant once we start executing the start point instead of the abort handler and I will add a note here about that (modulo the register state we will try to preserve).

Reply:
I always forget about push/pop being instructions that modify registers and memory, and clearly we remove memory-affecting instructions from RSEQ. You could add a note to this line to just say "e.g. push/pop instructions" to help people like me who did not think of this, otherwise it's totally fine with that knowledge in the front of my mind.

Reply:
I am adding the push example.